### DISCUSSION TITLE
Downloading doesn't work on Mac

### DISCUSSION THREAD

**Discussion by ethankrech** [Tyrrrz/YoutubeDownloader#565] [General]
Downloading the macos version, when you open the zip, all you get is a folder with a few hundred files inside, no app. I am on an M2 mac running the latest OS.

**Comment by rainymatch** [comment]
I guess most of the answer is obvious to you and many others, just writing it more in detail for those searching in the future: 

You will want to run the executable inside which is called "YoutubeDownloader"
But MacOS Gatekeeper, which is ensuirng that apps are from verified developers and free from malware, will complain that it is unknown piece of software and won't let you run it.
To open the executable, you can do right click > open, but the moment it starts loading the libraries, you will, once again, see some complaints from MacOS.

To bypass these restrictions, you can remove the quarantine attribute for all files in the folder using Terminal:

1) open Terminal in the folder you extracted those 263 files. 
To open terminal, you can use the spotlight search, CMD+Space, than type Terminal and enter... Next, navigate to the folder where you extracted the zip file. For example, if you placed the folder in the Applications folder, you would open run: 
```
cd /Applications/YoutubeDownloader
```
Here's how that works: `cd` stands for "change directory" and is used to navigate folders in Terminal. If your folder names contain spaces, you need to enclose the path in quotes, like this: 
```
cd '/Users/My User/Downloads/YoutubeDownloader' 
```

2) Remove the Quarantine Attribute
So now you are in the right folder and can run the following command: 
```sh
xattr -rd com.apple.quarantine ./*
```
`./*` means "all files in the current directory."

You could also just skip the navigation part in and run this command directly with the full path of the folder, from wherever: 
```sh
xattr -rd com.apple.quarantine /Applications/YoutubeDownloader/*
```

3) open the executable
So now you should be able to run the executable file `YoutubeDownloader` in the folder with you extracted files.
You may need to use right-click > open to start the applictaion. 
The program will run in Terminal, and the GUI will appear on your screen. You cannot close the Terminal window without also closing the GUI, as the program runs within the Terminal.

Hope it helps, have fun.
